### PR TITLE
Spring Boot Portfolio Project — Part 2

### PR DESCRIPTION
**Pull request by mgucma**

### CODE REVIEW

File: src/main/java/com/marek/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/books")
+public class BookController {
+    private BookService bookService;
```
Comment (MateuszRostek) [inline src/main/java/com/marek/onlinebookstore/controller/BookController.java]:
```suggestion
    private final BookService bookService;
```

File: src/main/java/com/marek/onlinebookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.marek.onlinebookstore.dto;
+
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    private String title;
+    private String author;
+    private String isbn;
+    private Double price;
```
Comment (MateuszRostek) [inline src/main/java/com/marek/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    private BigDecimal price;
```

**Review by MateuszRostek — CHANGES_REQUESTED** [review]
Where are changes in the `pom.xml` file? Dependencies for mapstruct, spring boot starter web and also one plugin + configuration changed in another one?

Also, you are missing `spring.jpa.open-in-view=false` in the `src/main/resources/application.properties`

**Comment by MateuszRostek** [comment]
Now I see that you have added these changes in the `pom.xml` in the previous Pull Request, next time **don't add more than the instruction is indicating, as this can lead to such problems**

File: src/main/java/com/marek/onlinebookstore/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (mgsorigo) [reply src/main/java/com/marek/onlinebookstore/dto/CreateBookRequestDto.java]: oki

File: src/main/java/com/marek/onlinebookstore/controller/BookController.java (hunk 1 above)
Comment (mgucma) [reply src/main/java/com/marek/onlinebookstore/controller/BookController.java]: Variable 'bookService' might not have been initialized ?

**Pull request merged**